### DISCUSSION TITLE
breadcrumb disappears on multiple refreshes

### DISCUSSION THREAD

**Discussion by SangeetaJ** [udayvunnam/xng-breadcrumb#24] [Q&A] [answered]
**Describe the bug**
When you refresh the page multiple times breadcrumb disappears. 
I'm using     **"xng-breadcrumb": "^4.0.0",**

This is what I get in console when breadcrumb doesn't show up : 
![image](https://user-images.githubusercontent.com/4082548/71171249-51673580-2283-11ea-94fe-2feb8d7ba6ba.png)

any idea what's wrong?

**Comment by udayvunnam** [comment, marked as answer]
@SangeetaJ I think I understood your problem. The breadcrumb which is set dynamically in component is lost and not shown when refreshed.

The problem is where you set the ```this.breadcrumbService.set('/path', 'label')```

- When you move from one route to another, we store all the dynamically provided labels and show them in Breadcrumb.
- But when do a refresh, you have to make sure the component where you set the label dynamically is loaded into the view.
- You can move up the logic ```this.breadcrumbService.set()``` to it's nearest parent which is always in the view no matter whichever child is rendered.

**Reply by CongLeVN** [reply]
I have moved up the logic to its nearest parent, even all of the ancestors but it still disappeared. Please tell me what is my wrong?

**Comment by SangeetaJ** [comment]
Hi @udayvunnam ,

That worked!

Thank you so much. Closing this issue.
